packages/flutter_tools/lib/src/asset.dart

```diff
@@ -94,7 +94,7 @@ class _ManifestAssetBundle implements AssetBundle {
       if (!directory.existsSync()) {
         return true; // directory was deleted.
       }
-      for (File file in directory.listSync()) {
+      for (File file in directory.listSync().whereType<File>()) {
```

Comment:
Why is this not a recursive listing?

Reply:
Wildcard assets aren't recursive

Reply:
Ahh, okay. Might be worth a comment here when you get a chance.